src/MessageTrait.php

```diff
@@ -196,21 +152,39 @@ private function setHeaders(array $headers): void
     }
 
     /**
-     * Trims whitespace from the header values.
+     * Make sure header has a non-empty string name and a sting value.
      *
-     * Spaces and tabs ought to be excluded by parsers when extracting the field value from a header field.
+     * Trims whitespace from the header values. Spaces and tabs ought to be
+     * excluded by parsers when extracting the field value from a header field.
      *
      * header-field = field-name ":" OWS field-value OWS
      * OWS          = *( SP / HTAB )
      *
-     * @param string[] $values Header values
-     *
-     * @return string[] Trimmed header values
-     *
      * @see https://tools.ietf.org/html/rfc7230#section-3.2.4
      */
-    private function trimHeaderValues(array $values): array
+    private function validateAndTrimHeader($header, $values): array
     {
+        if (!is_array($values)) {
+            $values = [$values];
+        } elseif (empty($values)) {
+            throw new \InvalidArgumentException('Header values must be strings, empty array given.');
+        } else {
+            // Non empty array
+            $values = array_values($values);
+        }
+
+        if (!is_string($header) || empty($header)) {
+            throw new \InvalidArgumentException('Header name must be strings');
+        }
+
+        foreach ($values as &$v) {
+            if (is_numeric($v)) {
+                $v = (string) $v;
+            } elseif (!is_string($v)) {
```

Comment:
Should we also check that the header value is allowed per RFC 7230? More specifically this means it consists of 0 or more `field-content` exclusively (IMHO we can ignore `obs-fold` as it was [deprecated in most use-cases](https://tools.ietf.org/html/rfc7230#section-3.2.4)):

```
field-value    = *( field-content / obs-fold )
field-content  = field-vchar [ 1*( SP / HTAB ) field-vchar ]
field-vchar    = VCHAR / obs-text
obs-text       = %x80-FF
VCHAR          = %x21-7E
```